Engine/ScriptAnalyzer.cs

```diff
@@ -1555,20 +1557,24 @@ public string Fix(string scriptDefinition, out bool fixesWereApplied)
                 throw new ArgumentNullException(nameof(scriptDefinition));
             }
 
-            Range updatedRange;
-            return Fix(new EditableText(scriptDefinition), null, out updatedRange, out fixesWereApplied).ToString();
+            ScriptBlockAst scriptAst = null;
+            Token[] scriptTokens = null;
+            return Fix(new EditableText(scriptDefinition), range: null, skipParsing: false, out _, out fixesWereApplied, ref scriptAst, ref scriptTokens).ToString();
         }
 
         /// <summary>
         /// Fix the violations in the given script text.
         /// </summary>
-        /// <param name="text">An object of type `EditableText` that encapsulates the script text to be fixed.</param>
+        /// <param name="text">An object of type <see cref="EditableText"/> that encapsulates the script text to be fixed.</param>
         /// <param name="range">The range in which the fixes are allowed.</param>
+        /// <param name="skipParsing">Whether to use the <paramref name="scriptAst"/> and <paramref name="scriptTokens"/> parameters instead of parsing the <paramref name="text"/> parameter.</param>
         /// <param name="updatedRange">The updated range after the fixes have been applied.</param>
-        /// <param name="updatedRange">Whether any warnings were fixed.</param>
+        /// <param name="fixesWereApplied">Whether any warnings were fixed.</param>
+        /// <param name="scriptAst">Optionally pre-parsed AST.</param>
+        /// <param name="scriptTokens">Optionally pre-parsed tokens.</param>
         /// <param name="skipVariableAnalysis">Whether to skip variable analysis.</param>
         /// <returns>The same instance of `EditableText` that was passed to the method, but the instance encapsulates the fixed script text. This helps in chaining the Fix method.</returns>
-        public EditableText Fix(EditableText text, Range range, out Range updatedRange, out bool fixesWereApplied, bool skipVariableAnalysis = false)
+        public EditableText Fix(EditableText text, Range range, bool skipParsing, out Range updatedRange, out bool fixesWereApplied, ref ScriptBlockAst scriptAst, ref Token[] scriptTokens, bool skipVariableAnalysis = false)
```

Comment:
Technically this is a breaking change as well but I think this class shouldn't have been public but rather internal in the first place. Again, we could still provide a wrapping layer for legacy purposed but it'd just leave the change as-is.